flytekit/clis/sdk_in_container/run.py

```diff
@@ -775,6 +819,76 @@ def get_command(self, ctx, name):
         )
 
 
+class YamlFileReadingCommand(click.RichCommand):
+    def __init__(
+        self,
+        name: str,
+        params: typing.List[click.Option],
+        help: str,
+        callback: typing.Callable = None,
+    ):
+        params.append(
+            click.Option(
+                ["--inputs-file"],
+                required=False,
+                type=click.Path(exists=True, dir_okay=False, resolve_path=True),
+                help="Path to a YAML | JSON file containing inputs for the workflow.",
+            )
+        )
+        super().__init__(name=name, params=params, callback=callback, help=help)
+
+    def parse_args(self, ctx: Context, args: t.List[str]) -> t.List[str]:
+        inputs = {}
+        if "--inputs-file" in args:
+            idx = args.index("--inputs-file")
+            args.pop(idx)
+            f = args.pop(idx)
+            with open(f, "r") as f:
+                try:
+                    inputs = yaml.safe_load(f)
+                except yaml.YAMLError as e:
+                    yaml_e = e
+                    try:
+                        inputs = json.load(f)
+                    except json.JSONDecodeError as e:
+                        raise click.BadParameter(
+                            message=f"Could not load the inputs file. Please make sure it is a valid JSON or YAML file."
+                            f"\n json error: {e},"
+                            f"\n yaml error: {yaml_e}",
+                            param_hint="--inputs-file",
+                        )
+        elif not sys.stdin.isatty():
+            f = sys.stdin.read()
+            if f != "":
+                try:
+                    inputs = yaml.safe_load(f)
+                except yaml.YAMLError as e:
+                    yaml_e = e
+                    try:
+                        inputs = json.loads(f)
+                    except json.JSONDecodeError as e:
+                        raise click.BadParameter(
+                            message=f"Could not load the inputs file. Please make sure it is a valid JSON or YAML file."
+                            f"\n json error: {e},"
+                            f"\n yaml error: {yaml_e}",
+                            param_hint="--inputs-file",
```

Comment:
Could we create a common function to read json/yaml file for `stdin` and `--input-file`?

Reply:
@pingsutw A common function `load_inputs` is added for parsing inputs from `stdin` and `--inputs-file`. Hope it looks good to us ~